src/posthog-core.ts

```diff
@@ -1911,17 +1911,20 @@ export class PostHog {
      *     });
      *
      * @param {Object} [config] A dictionary of config options to override
-     * @param {string} [config.capture_event_name=$opt_in] Event name to be used for capturing the opt-in action
+     * @param {string} [config.capture_event_name=$opt_in] Event name to be used for capturing the opt-in action. Set to `null` or `false` to skip capturing the optin event
      * @param {Object} [config.capture_properties] Set of properties to be captured along with the opt-in action
      */
     opt_in_capturing(options?: {
-        captureEventName?: string /** event name to be used for capturing the opt-in action */
+        captureEventName?: string | null | false /** event name to be used for capturing the opt-in action */
         captureProperties?: Properties /** set of properties to be captured along with the opt-in action */
     }): void {
         this.consent.optInOut(true)
         this._sync_opt_out_with_persistence()
 
-        // TODO: Do we need it to be sent instantly?
+        if (!isUndefined(options?.captureEventName) && !options.captureEventName) {
```

Comment:
```suggestion
        if (isNull(options?.captureEventName) || (isBoolean(options?.captureEventName) && options?.captureEventName === false)) {
```

or something like that... it's longer but a little easier to parse IMHO